src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -91,24 +92,69 @@ public static CodeWriter WriteFieldDeclaration(this CodeWriter writer, FieldDecl
 
         public static CodeWriter.CodeWriterScope WriteMethodDeclaration(this CodeWriter writer, MethodSignatureBase methodBase, params string[] disabledWarnings)
         {
-            foreach (var disabledWarning in disabledWarnings)
+            WriteDisableWarnings(writer, disabledWarnings);
+
+            writer.Append($"{methodBase.Modifiers} ");
+            if (methodBase is MethodSignature method)
             {
-                writer.Line($"#pragma warning disable {disabledWarning}");
+                if (method.ReturnType != null)
+                {
+                    writer.Append($"{method.ReturnType} ");
+                }
+                else
+                {
+                    writer.AppendRaw("void ");
+                }
             }
 
+            return WriteMethodDeclarationParameters(writer, methodBase, disabledWarnings, methodBase.Name);
+        }
+
+        public static CodeWriter.CodeWriterScope WriteMethodDeclaration(this CodeWriter writer, MethodSignatureBase methodBase, bool isAsync, params string[] disabledWarnings)
```

Comment:
With this consolidation, would it be possible to merge this method with the one above?

Reply:
This one dynamically adds virtual, async, wraps reaponse in Task etc.  The overload above is called all  over the place where people are doing that extra work themselves.  We can take that as a task later to update all the dataplane calls into this method to no longer do any of that logic themselves.